Modules/mmapmodule.c

```diff
@@ -682,6 +682,60 @@ mmap__exit__method(PyObject *self, PyObject *args)
     return _PyObject_CallMethodId(self, &PyId_close, NULL);
 }
 
+static PyObject *
+mmap__repr__method(PyObject *self)
+{
+    PyObject *repr;
+    mmap_object *m_obj;
+    m_obj = (mmap_object *) self;
+
+    const char *reprfmt;
+    const char *access_str;
+
+    switch (m_obj->access) {
+        case ACCESS_DEFAULT:
+            access_str = "ACCESS_DEFAULT";
+            break;
+
+        case ACCESS_READ:
+            access_str = "ACCESS_READ";
+            break;
+
+        case ACCESS_WRITE:
+            access_str = "ACCESS_WRITE";
+            break;
+
+        case ACCESS_COPY:
+            access_str = "ACCESS_COPY";
+            break;
+
+        default:
+            Py_UNREACHABLE();
+    }
+
+    const char *tp_name = Py_TYPE(self)->tp_name;
+
+#ifdef MS_WINDOWS
+    if (m_obj->map_handle == NULL)
+#elif defined(UNIX)
+    if (m_obj->data == NULL)
+#endif
+    {
+        reprfmt = "<%s closed=True access=%s>";
+        repr = PyUnicode_FromFormat(reprfmt, tp_name, access_str);
+    }
+    else {
+        PyObject *length = PyLong_FromSize_t(m_obj->size);
+        PyObject *pos = PyLong_FromSize_t(m_obj->pos);
+        PyObject *offset = PyLong_FromSize_t((Py_ssize_t)m_obj->offset);
+        reprfmt = "<%s closed=False access=%s length=%R pos=%R offset=%R>";
+        repr = PyUnicode_FromFormat(reprfmt, tp_name, access_str,
+                                    length, pos, offset);
```

Comment:
It should be `reprfmt = "<%s closed=False, pos=%zd, length=%zd, offset=%ld/%lld, access=%s>", then `PyUnicode_FromFormat(reprfmt, tp_name, pos, length, offset, access_str`. You don't need to call the `PyLong_FromSize_t` yourself. More, I suggest add comma between properties. And be careful to decide which format character `offset` uses, you'd better don't cast anyway, you need to decide using macros, check `new_mmap_object`.

Reply:
Okay, but for the `,` separator, I'm not sure yet. In fact there're both styles(`<a b c>` like [fileio](https://github.com/python/cpython/blob/master/Modules/_io/fileio.c#L1080) and `<a, b, c>`) in `repr`.

Reply:
For `offset`, I do want to use `PyLong_FromOff_t` because `off_t` type is a bit tricky in cross-platform issues. 
If I include another header file for only one function, it's much too cumbersome; but if I don't, I have to do what that've been done in [_iomodule.h](https://github.com/python/cpython/blob/master/Modules/_io/_iomodule.h#L105) once again.

Reply:
For comma it's just preference, I prefer it. There is no `PyLong_FromOff_t` and don't include _iomodule.h, usually we don't include dependency between modules. Use macros to determine the right format character, refer (new_mmap_object)[https://github.com/python/cpython/blob/master/Modules/mmapmodule.c#L1040] please. Don't call `PyLong_From*` yourself, just use `PyUnicode_FromFormat`.